Document Enterprise API (#2595)
pomerium/pomerium: docs/enterprise/api.md

---
title: API
lang: en-US
meta:
  - name: keywords
    content: pomerium identity-access-proxy oidc reverse-proxy enterprise console api python go
---

# Enterprise Console API

> The Pantheon Enterprise Console supports programmatic interaction through an API. This page covers enabling and authenticating to the API.

Reviewer: ```suggestion
The Pomerium Enterprise Console supports programmatic interaction through an API. This page covers enabling and authenticating to the API.
```

Author: Whoooopsie!

 - The enterprise console service is encrypted. Review the [tls-*] keys for more information.

## Configure a New Route

1. We suggest configuring the route for API access in the open-source Pomerium. That way changes made through the API that might break access to the console GUI will not break access to the API route.

	```yaml
	- from: https://console-api.pomerium.localhost.io
		to: https://pomerium-console-domain-name:8702
		pass_identity_headers: true
		allow_any_authenticated_user: true
		tls_custom_ca_file: /path/to/rootCA.pem # See https://www.pomerium.com/reference/#tls-custom-certificate-authority
	```

1. You must also update the `audience` key to include the new route's `from` value:

	```yaml
	audience: "console.pomerium.localhost.io,console-api.pomerium.localhost.io"
	```

	If you're running Pomerium Enterprise as a system service, restart the daemon.

## Create a Service Account

1. In the enterprise Console under **Configure -> Service Accounts**, Click **+ Add Service Account**. You can choose an existing user for the service account to impersonate, or create a new user. Note that a new user will not be synced to your IdP.

> 1. The Enterprise Console will display the service account token. Be sure to store it securely not, as you cannot view it again after this point.

Reviewer: ```suggestion
1. The Enterprise Console will display the service account token. Be sure to store it securely now, as you cannot view it again after this point.
```

> 1. Add the service account to the Namespaces you want it to access, with the correct role based on the actions it will perform.

Reviewer: ```suggestion
1. Grant the service account the appropriate role for access to various Namespaces
```

Reviewer: Soft suggestion that I don't love.  But I think we should be careful about phrasing and the difference between creating a service account in a namespace and granting it permissions on that namespace.  "add to a namespace" is a bit ambiguous.

Author: Agreed that my wording was not great. How about:

```suggestion
1. Grant the service account the appropriate [role](/enterprise/concepts.md#rbac-for-enterprise-console-users) in the Namespace(s) it will operate in.
```

Author: ```suggestion
1. Grant the service account the appropriate [role](/enterprise/concepts.md#rbac-for-enterprise-console-users) on the Namespace(s) it will operate against.

```

::: tip
Service accounts created in any Namespace other than **Global** will include a reference to that Namespace ID. You must specify the entire user ID (i.e. `design-api@bff1bea6-a3d6-232d-812c-b4fd8e26d72e.pomerium`) when using the service account.
:::

## Install The Library

:::: tabs
::: tab Python
```bash
pip3 install git+ssh://git@github.com/pomerium/enterprise-client-python.git
```
:::
::: tab Go
```bash

> go get github.com:pomerium/enterprise-client-go.git

Reviewer: ```suggestion
go get github.com/pomerium/enterprise-client-go
```
